girder_worker_utils/json.py

```diff
@@ -1,23 +1,49 @@
 from __future__ import absolute_import
 
 from functools import wraps
+from collections import OrderedDict
+import importlib
 
-import json
 
+try:
+    from kombu.utils import json
+except ImportError:
+    import json
 
-_hook_registry = {}
 
+# Singleton/ClassVariableSingleton.py
 
-def register_hook(name, func, *args, **kwargs):
-    _hook_registry[name] = func, args, kwargs
+class Hook(object):
+    """Deserialize objects with __class_hint__ attributes.
 
+    This class abstracts over the process of turning a json dictionary
+    into an instantiated object.  It does this by accepting a string
+    keyword argument for the module and function that constructs the
+    object. You may optionally pass in class if the object constructor
+    is a classmethod.
 
-def hook(name, *args, **kwargs):
-    def decorator(func):
-        register_hook(name, func, *args, **kwargs)
-        return func
-    return decorator
+    """
+    def __init__(self, func=None, module=None, cls=None, **kwargs):
```

Comment:
Remove `=None` from `func` and `module` so an error is thrown if they are not provided.